src/Fmt_ast.ml

```diff
@@ -1529,10 +1530,9 @@ and fmt_expression c ?(box = true) ?pro ?epi ?eol ?parens ?(indent_wrap = 0)
                              | _ -> None )
                            xbody )
                    $ fmt_or_k c.conf.indicate_multiline_delimiters
-                       (fmt_or_k
-                          (Location.is_single_line pexp_loc c.conf.margin)
-                          (fmt ")") (fits_breaks ")" "@ )"))
-                       (fits_breaks ")" "@,)") )
+                       (fmt_or_k fit (fmt ")")
+                          (fits_breaks ~force_fit_if:fit ")" "@ )"))
```

Comment:
Is `fits_breaks` necessary if we have `fit` ?

Reply:
Yes, any other thing I tried just caused more regressions (I didn't find a way to get rid of the `Location.is_single_line` either).

Reply:
I have no regressions with 
```patch
--- a/src/Fmt_ast.ml
+++ b/src/Fmt_ast.ml
@@ -1530,9 +1530,8 @@ and fmt_expression c ?(box = true) ?pro ?epi ?eol ?parens ?(indent_wrap = 0)
                              | _ -> None )
                            xbody )
                    $ fmt_or_k c.conf.indicate_multiline_delimiters
-                       (fmt_or_k fit (fmt ")")
-                          (fits_breaks ~force_fit_if:fit ")" "@ )"))
-                       (fits_breaks ~force_fit_if:fit ")" "@,)") )
+                       (fmt_or_k fit (fmt ")") (fmt_or fit ")" "@ )"))
+                       (fmt_or fit ")" "@,)") )
                $ fmt_atrs ))
       | ( lbl
         , ( { pexp_desc= Pexp_function [{pc_lhs; pc_guard= None; pc_rhs}]
```

Reply:
With this change I get the following diff:

```ocaml
diff --git a/src/Ast.ml b/src/Ast.ml
index 0031f7f..8b0ef74 100644
--- a/src/Ast.ml
+++ b/src/Ast.ml
@@ -805,8 +805,7 @@ end = struct
               || loop e
           | Pcf_method (_, _, Cfk_concrete _) -> false
           | Pcf_constraint (t1, t2) -> t1 == typ || t2 == typ
-          | Pcf_initializer _ | Pcf_attribute _ | Pcf_extension _ -> false
-      )
+          | Pcf_initializer _ | Pcf_attribute _ | Pcf_extension _ -> false )
     in
     let check_class_type l =
       List.exists l ~f:(fun {pci_expr= {pcty_desc}; pci_params} ->
diff --git a/src/Conf.ml b/src/Conf.ml
index 2f218be..d7da1a0 100644
--- a/src/Conf.ml
+++ b/src/Conf.ml
@@ -1840,8 +1840,7 @@ let action =
     Inplace
       (List.map !inputs ~f:(fun file ->
            let name = Option.value !name ~default:file in
-           {kind= kind_of file; name; file; conf= build_config ~file:name}
-       ))
+           {kind= kind_of file; name; file; conf= build_config ~file:name} ))
   else if !check then
     Check
       (List.map !inputs ~f:(fun file ->
diff --git a/src/Fmt_ast.ml b/src/Fmt_ast.ml
index 3bc41a8..d0ee0f1 100644
--- a/src/Fmt_ast.ml
+++ b/src/Fmt_ast.ml
@@ -2271,8 +2270,8 @@ and fmt_class_structure c ~ctx ?ext self_ fields =
         ( fmt "object"
         $ fmt_extension_suffix c ext
         $ opt self_ (fun self_ ->
-              fmt "@;" $ wrap "(" ")" (fmt_pattern c (sub_pat ~ctx self_))
-          ) )
+              fmt "@;" $ wrap "(" ")" (fmt_pattern c (sub_pat ~ctx self_)) )
+        )
     $ cmts_after_self
     $ ( match fields with
       | ({pcf_desc= Pcf_attribute a}, _) :: _
@@ -2318,8 +2317,8 @@ and fmt_class_signature c ~ctx ~parens ?ext self_ fields =
                $ fmt_extension_suffix c ext
                $ opt self_ (fun self_ ->
                      fmt "@;"
-                     $ wrap "(" ")" (fmt_core_type c (sub_typ ~ctx self_))
-                 ) )
+                     $ wrap "(" ")" (fmt_core_type c (sub_typ ~ctx self_)) )
+               )
            $ cmts_after_self
            $ ( match fields with
              | ({pctf_desc= Pctf_attribute a}, _) :: _
@@ -3793,8 +3792,7 @@ and fmt_structure c ctx itms =
         fmt_if_k (not first) (fmt_or break_struct "@\n" "@ ")
         $ maybe_disabled c itm.pstr_loc []
           @@ fun c ->
-          fmt_structure_item c ~last:(last && last_grp) (sub_str ~ctx itm)
-    )
+          fmt_structure_item c ~last:(last && last_grp) (sub_str ~ctx itm) )
```

Reply:
Sorry I forgot to test with `make fmt`.